JitsiConference.js

```diff
@@ -1132,97 +1125,119 @@ JitsiConference.prototype.onIncomingCall
-         *  would clash with the one firefox generates (since, unlike chrome,
-         *  the stream is still attached to the peer connection) and causes
-         *  problems between sdp-interop and trying to keep the ssrcs
-         *  consistent
-         */
-
-        if (localTrack.isVideoTrack()
-                && localTrack.isMuted()
-                && !RTCBrowserType.isFirefox()) {
-            /**
-             * Handles issues when the stream is added before the peerconnection
-             * is created. The peerconnection is created when second participant
-             * enters the call. In that use case the track doesn't have
-             * information about it's ssrcs and no jingle packets are sent. That
-             * can cause inconsistent behavior later.
-             *
-             * For example:
-             * If we mute the stream and than second participant enter it's
-             * remote SDP won't include that track. On unmute we are not sending
-             * any jingle packets which will brake the unmute.
-             *
-             * In order to solve issues like the above one here we have to
-             * generate the ssrc information for the track .
-             */
-            localTrack._setSSRC(this._generateNewStreamSSRCInfo());
-            ssrcInfo = {
-                mtype: localTrack.getType(),
-                type: 'addMuted',
-                ssrcs: localTrack.ssrc.ssrcs,
-                groups: localTrack.ssrc.groups,
-                msid: localTrack.initialMSID
-            };
-        }
-        try {
-            this._addLocalStream(
-                localTrack.getOriginalStream(),
-                () => {
-                    // There is nothing to be done after the stream is added.
-                },
-                () => {
-                    // TODO: Maybe report the error to the user?
-                },
-                ssrcInfo,
-                true /* don't modify SSRCs */);
-        } catch (e) {
-            GlobalOnErrorHandler.callErrorHandler(e);
-            logger.error(e);
-        }
-    });
+    // Add local tracks to the session
+    try {
+        jingleSession.addLocalTracks(this.getLocalTracks()).then(() => {
+            jingleSession.acceptOffer(jingleOffer, null,
+                error => {
+                    GlobalOnErrorHandler.callErrorHandler(error);
+                    logger.error(
+                        'Failed to accept incoming Jingle session', error);
+                }
+            );
+
+            // Start callstats as soon as peerconnection is initialized,
+            // do not wait for XMPPEvents.PEERCONNECTION_READY, as it may never
+            // happen in case if user doesn't have or denied permission to
+            // both camera and microphone.
+            this.statistics.startCallStats(jingleSession);
+            this._startRemoteStats();
+        });
+    } catch (e) {
```

Comment:
I have no idea. I imagine this try catch was introduced to protect from breaking strophe packet handler when something goes wrong, but we would have to ask the original author. Not sure what to do about this one. Remove or add a comment ?